src/_objectProps.ts

```diff
@@ -0,0 +1,42 @@
+import { Simplify } from './type-fest/simplify';
+
+export type AnyValidObjectKey = string | number | symbol;
+export type GenericObject = Record<AnyValidObjectKey, unknown>;
```

Comment:
**suggestion:** we have the same type in `isPlainObject`; so maybe this should be called `PlainObject`?